pallets/parachain-staking/src/inflation.rs

```diff
@@ -0,0 +1,215 @@
+    let circulating = <T as Config>::Currency::total_issuance();
+    Range {
+        min: round.min * circulating,
+        ideal: round.ideal * circulating,
+        max: round.max * circulating,
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Eq, PartialEq, Clone, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+pub struct InflationInfo<Balance> {
+    /// Staking expectations
+    pub expect: Range<Balance>,
+    /// Annual inflation range
+    pub annual: Range<Perbill>,
+    /// Round inflation range
+    pub round: Range<Perbill>,
+}
+
+impl<Balance> InflationInfo<Balance> {
+    pub fn new<T: Config>(
+        annual: Range<Perbill>,
+        expect: Range<Balance>,
+    ) -> InflationInfo<Balance> {
+        InflationInfo {
+            expect,
+            annual,
+            round: annual_to_round::<T>(annual),
+        }
+    }
+    /// Set round inflation range according to input annual inflation range
+    pub fn set_round_from_annual<T: Config>(&mut self, new: Range<Perbill>) {
+        self.round = annual_to_round::<T>(new);
+    }
+    /// Reset round inflation rate based on changes to round length
+    pub fn reset_round(&mut self, new_length: u32) {
+        let periods = BLOCKS_PER_YEAR / new_length;
+        self.round = perbill_annual_to_perbill_round(self.annual, periods);
+    }
+    /// Set staking expectations
+    pub fn set_expectations(&mut self, expect: Range<Balance>) {
+        self.expect = expect;
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    fn mock_annual_to_round(annual: Range<Perbill>, rounds_per_year: u32) -> Range<Perbill> {
+        perbill_annual_to_perbill_round(annual, rounds_per_year)
+    }
+    fn mock_round_issuance_range(
+        // Total circulating before minting
+        circulating: u128,
+        // Round inflation range
+        round: Range<Perbill>,
+    ) -> Range<u128> {
+        Range {
+            min: round.min * circulating,
+            ideal: round.ideal * circulating,
+            max: round.max * circulating,
+        }
+    }
+    #[test]
+    fn simple_issuance_conversion() {
+        // 5% inflation for 10_000_0000 = 500,000 minted over the year
+        // let's assume there are 10 periods in a year
+        // => mint 500_000 over 10 periods => 50_000 minted per period
+        let expected_round_issuance_range: Range<u128> = Range {
+            min: 48_909,
+            ideal: 48_909,
+            max: 48_909,
+        };
+        let schedule = Range {
+            min: Perbill::from_percent(5),
+            ideal: Perbill::from_percent(5),
+            max: Perbill::from_percent(5),
+        };
+        assert_eq!(
+            expected_round_issuance_range,
+            mock_round_issuance_range(10_000_000, mock_annual_to_round(schedule, 10))
+        );
+    }
+    #[test]
+    fn range_issuance_conversion() {
+        // 3-5% inflation for 10_000_0000 = 300_000-500,000 minted over the year
```

Comment:
something is off with these numbers

Reply:
Not sure what you mean specifically other than the missing _ on ideal, but this is testcode anyway

Reply:
5% of 10_000_0000 is 5_000_000 nor 500_000
Also not sure where the 10_000_0000 is coming from

Reply:
Same on line 160

Reply:
10mil is GLMR total issuance
5% is 500k, 5mil is 50%

~39k is ~ 400k ( 4% of issuance ) divided by 10 periods to get per-round expected inflation.

These tests are just for the conversion so we don't have to change the numbers

Reply:
10_000_0000 is 100mil not 10mil ...

Reply:
yeah, it's a typo in the comment i imagine.
that `_0000` should have been `_000`